test/post_processor/test_post_processor.py

```diff
@@ -8,42 +8,157 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Tests for the MultiQubitPOVM class."""
+"""Tests for the POVMPostProcessor class."""
 
 from unittest import TestCase
 
-from povm_toolbox.library import ClassicalShadows
+import numpy as np
+from povm_toolbox.library import ClassicalShadows, LocallyBiasedClassicalShadows
 from povm_toolbox.post_processor import POVMPostProcessor
+from povm_toolbox.quantum_info.product_dual import ProductDUAL
 from povm_toolbox.sampler import POVMSampler
 from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
 from qiskit.primitives import StatevectorSampler as Sampler
 from qiskit.quantum_info import SparsePauliOp
 
 
 class TestPostProcessor(TestCase):
-    """TODO."""
+    """Test the methods and attributes of the :class:`.POVMPostProcessor class`."""
 
-    def test_init(self):
-        """TODO."""
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
```

Comment:
Why are these changes necessary?

Reply:
I added an `__init__` to the test to define `self.pub_result` which is re-used multiple times in different tests. This is to avoid having to define a circuit, sampler, etc and run a job for each test in the file.

Reply:
Yes, having the `__init__` is fine. I have just not seen this `methodName` before and am wondering why/if it is relevant/needed

Reply:
I don't remember where I saw I was supposed to add the argument `methodName`, but I get an error if I remove it.
I could also write 
```python
def __init__(self, *args, **kwargs) -> None:
    super().__init__(*args, **kwargs)
```
alternatively, it also works. What do you think ?

Reply:
Ah I remember again! We normally leave `__init__` unchanged and instead overwrite the `setUp` method (https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp).
So you would do:
```python
class TestClass(unittest.TestCase):
    def setUp(self) -> None:
        super().setUp()

        # whatever else you want to do
```

Reply:
Okay I see, I'll make the change!